src/components/sequencing/Sequences.svelte

```diff
@@ -0,0 +1,172 @@
+
+        return actionsDiv;
+      },
+      cellRendererParams: {
+        deleteSequence,
+        editSequence,
+      } as CellRendererParams,
+      field: 'actions',
+      headerName: '',
+      resizable: false,
+      sortable: false,
+      suppressAutoSize: true,
+      suppressSizeToFit: true,
+      width: 55,
+    },
+  ];
+
+  let filterText: string = '';
+  let filteredSequences: UserSequence[] = [];
+  let selectedSequence: UserSequence | null = null;
+  let selectedSequenceSeqJson: string = 'No Sequence Selected';
+  let sortedSequences: UserSequence[] = [];
+
+  $: filteredSequences = $userSequences.filter(sequence => {
+    const filterTextLowerCase = filterText.toLowerCase();
+    const includesId = `${sequence.id}`.includes(filterTextLowerCase);
+    const includesName = sequence.name.toLocaleLowerCase().includes(filterTextLowerCase);
+    return includesId || includesName;
+  });
+  $: sortedSequences = filteredSequences.sort((a, b) => compare(a.id, b.id));
+  $: if (selectedSequence !== null) {
+    const found = $userSequences.findIndex(sequence => sequence.id === selectedSequence.id);
+    if (found === -1) {
+      selectedSequence = null;
+    }
+  }
+
+  async function deleteSequence({ id }: UserSequence) {
+    const success = await effects.deleteUserSequence(id);
+
+    if (success) {
+      sortedSequences = sortedSequences.filter(sequence => sequence.id !== id);
+
+      if (id === selectedSequence?.id) {
+        selectedSequence = null;
+      }
+    }
+  }
+
+  async function getUserSequenceSeqJson(): Promise<void> {
+    if (selectedSequence !== null) {
+      selectedSequenceSeqJson = 'Generating Seq JSON...';
+      selectedSequenceSeqJson = await effects.getUserSequenceSeqJson(
+        selectedSequence.authoring_command_dict_id,
+        selectedSequence.definition,
+      );
+    }
+  }
+
+  function editSequence({ id }: UserSequence) {
+    goto(`${base}/sequencing/edit/${id}`);
+  }
+
+  async function toggleSequence(event: CustomEvent<DataGridRowSelection<UserSequence>>) {
+    const { detail } = event;
+    const { data: clickedSequence, isSelected } = detail;
+
+    if (isSelected) {
+      selectedSequence = clickedSequence;
+      await getUserSequenceSeqJson();
+    }
+  }
+</script>
+
+<CssGrid bind:columns={$userSequencesColumns}>
+  <Panel>
+    <svelte:fragment slot="header">
+      <Chip>Sequences</Chip>
+
+      <Input>
+        <input
+          bind:value={filterText}
+          class="st-input"
+          placeholder="Filter sequences"
+          style="max-width: 300px; width: 100%;"
+        />
+      </Input>
+
+      <div class="right">
+        <button class="st-button secondary ellipsis" on:click={() => goto(`${base}/sequencing/new`)}> New </button>
+      </div>
+    </svelte:fragment>
+
+    <svelte:fragment slot="body">
+      {#if sortedSequences.length}
+        <DataGrid {columnDefs} rowData={sortedSequences} rowSelection="single" on:rowSelected={toggleSequence} />
```

Comment:
Would it be possible to auto resize the columns in the sequence table? I'm finding that by default the sidebar is too narrow to see any of the values in the table so I have to expand the sidebar but the grid doesn't auto resize. 
<img width="1139" alt="image" src="https://user-images.githubusercontent.com/4419607/187764672-3466db37-067c-4c7e-b926-122a10a9965e.png">

Reply:
@duranb Is there an easy way to auto resize the data grid columns?

Reply:
Programatically, yes it's pretty easy. DataGrid already autosizes to fit/shrink on load. We'd just need to expose that function to the parent component. The harder question is how to handle it. Which columns get resized? If a user changes the columns to their preferred width, and then resizes, do we just blow away their changes?

Reply:
Thanks - will make a follow up task for this at some point